ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java

```diff
@@ -395,6 +414,71 @@ public String createZoneSQL() {
     return createZoneReq;
   }
 
+  /**
+   * Check if we have an opened transaction and begin a new one if not.
+   */
+  protected void checkBeginTx() {
+    if (tx == null) {
+      tx = ignite.transactions().begin();
+    }
+  }
+
+  /**
+   * Check whether the required number of operations has accumulated in the current transaction,
+   * and if so, commit this transaction.
+   */
+  protected void checkEndTx() throws TransactionException {
+    if (currOpsInTx >= txOps) {
+      commitTx();
+    }
+  }
+
+  /**
+   * Do commit current transaction.
+   */
+  protected void commitTx() throws TransactionException {
+    if (tx != null) {
+      tx.commit();
+      tx = null;
+      currOpsInTx = 0;
+    }
+  }
+
+  /**
+   * Do rollback current transaction.
+   */
+  protected void rollbackTx() throws TransactionException {
+    if (tx != null) {
+      tx.rollback();
+      tx = null;
+      currOpsInTx = 0;
+    }
+  }
+
+  /**
+   * If parameter 'txops' > 0,
+   * then perform the operation in transaction
+   *      (automatically begin transaction and commit it after 'txops' operations performed),
+   * else perform the operation without transaction.
+   *
+   * @param operation Operation that produces a result.
+   * @return operation call result.
+   */
+  protected <R> R wrapWithTx(Callable<R> operation) throws Exception {
+    if (!isWrapOpsToTx) {
+      return operation.call();
+    } else {
+      checkBeginTx();
+
+      R result = operation.call();
+      currOpsInTx++;
+
+      checkEndTx();
+
+      return result;
```

Comment:
Could we wrap this block into `try {} catch (TransactionException) {rollback; throw}` statement? It should remove some duplication from the client side.

Reply:
Yep, good idea